benchmarks/rigid_shear/plugin/rigid_shear.cc

```diff
@@ -64,25 +65,56 @@ namespace aspect
      */
     namespace AnalyticSolutions
     {
+      double
+      phase_function(const double t)
+      {
+        return std::exp(t)-1;
+      }
+
+      template<int dim>
+      double
+      density(const Point<dim> &p,
+              const double t,
+              const bool transient)
+      {
+        const double pi = numbers::PI;
+        const double phase = (transient == true) ? phase_function(t) : 0.0;
+
+        return std::sin(pi*(p[0]-phase)) * std::sin(pi*p[1]) + 2.0;
+      }
+
       /**
        * The exact solution for the Rigid Shear benchmark.
        */
       template<int dim>
       class FunctionRigidShear : public Function<dim>
       {
         public:
-          FunctionRigidShear(unsigned int n_components) : Function<dim>(n_components) {}
+          FunctionRigidShear(const unsigned int n_components,
+                             const bool transient)
+            : Function<dim>(n_components),
+              transient_(transient) {}
 
           virtual void vector_value(const Point<dim> &p,
                                     Vector<double> &values) const
           {
             const double pi = numbers::PI;
-            values[0] = std::sin(pi*p[0]) * std::cos(pi*p[1]);
-            values[1] = -std::cos(pi*p[0]) * std::sin(pi*p[1]);
-            values[2] = 2.0 * pi * std::cos(pi*p[0]) * std::cos(pi*p[1]);
-            values[4] = std::sin(pi*p[0]) * std::sin(pi*p[1]);
+            const double t = this->get_time();
+            const double phase = (transient_ == true) ? phase_function(t) : 0.0;
+
+            values[0] = std::sin(pi*(p[0]-phase)) * std::cos(pi*p[1]);
+
+            if (transient_ == true)
+              values[0] += std::exp(t);
+
+            values[1] = -std::cos(pi*(p[0]-phase)) * std::sin(pi*p[1]);
+            values[2] = 2.0 * pi * std::cos(pi*(p[0]-phase)) * std::cos(pi*p[1]);
+            values[4] = density(p,t,transient_);
             return;
           }
+
+        private:
+          bool transient_;
```

Comment:
Not a big fan of name_. It should also work if you use 
```suggestion
          bool transient;
```
(even if the constructor argument has the same name)